test/DukeTest.java

```diff
@@ -0,0 +1,90 @@
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+import duke.ui.UI;
+import org.junit.jupiter.api.*;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+public class DukeTest {
+
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private final ByteArrayOutputStream errContent = new ByteArrayOutputStream();
+    private final PrintStream originalOut = System.out;
+    private final PrintStream originalErr = System.err;
+
+    public static TaskList shelf;
+    public static Storage storage;
+    public static Parser parser;
+    public static UI ui;
+    public static DateTimeFormatter formatter;
+
+    @BeforeAll
+    public static void setUp() {
+        File f = new File("D:\\24092014\\Joven\\UNI STUFF\\CS2103\\IP\\task.txt");
+        storage = new Storage(f);
+        shelf = new TaskList(storage.loadFile());
+        ui = new UI(shelf, storage);
+        parser = new Parser(ui);
+        formatter = DateTimeFormatter.ofPattern("MMM d yyyy HH:mm");
+    }
+
+    @BeforeEach
+    public void setUpStreams() {
+        System.setOut(new PrintStream(outContent));
+        System.setErr(new PrintStream(errContent));
+    }
+
+    @AfterEach
+    public void restoreStreams() {
+        System.setOut(originalOut);
+        System.setErr(originalErr);
+    }
+
+    @Test
+    public void testBye() {
+        ui.replyBye();
+        assertEquals("CYA PAL. Hope to see you again!\r\n", outContent.toString());
+    }
+
+    @Test
+    public void testAddDeadLine_deadlineInput_success() {
+        try {
+            ui.addDeadline("finish homework!", "2020-10-02 11:44");
+            String currentTime = LocalDateTime.now().format(formatter);
+            assertEquals("Got it. I've added this task: \r\n  [D][-] finish homework! [created on " +
```

Comment:
Perhaps the + sign can be moved to the current second line of String to be consistent with the coding standards?